spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -253,51 +250,39 @@ function initUpgradeHandshake(
     // new order must be supported by the new connection
     abortTransactionUnless(isSupported(proposedConnection, proposedUpgradeFields.ordering))
 
-    // either timeout height or timestamp must be non-zero
-    abortTransactionUnless(timeout.timeoutHeight != 0 || timeout.timeoutTimestamp != 0)
-
-    // get last packet sent on channel and set it in the upgrade struct
-    // last packet sent is the nextSequenceSend on the channel minus 1
-    lastPacketSendSequence = provableStore.get(nextSequenceSendPath(portIdentifier, channelIdentifier)) - 1
+    // lastPacketSent and timeout will be filled when we move to FLUSHING
     upgrade = Upgrade{
         fields: proposedUpgradeFields,
-        timeout: timeout,
-        lastPacketSent: lastPacketSendSequence,
     }
 
     // store upgrade in public store for counterparty proof verification
     provableStore.set(channelUpgradePath(portIdentifier, channelIdentifier), upgrade)
 
     currentChannel.sequence = currentChannel.sequence + 1
-    currentChannel.state = INITUPGRADE
     provableStore.set(channelPath(portIdentifier, channelIdentifier), channel)
     return currentChannel.sequence
 }
 ```
 
-`startFlushUpgradeHandshake` will set the counterparty last packet send and continue blocking the upgrade from continuing until all in-flight packets have been flushed. When the channel is in blocked mode, any packet receive above the counterparty last packet send will be rejected. It will verify the upgrade parameters and set the channel state to one of the flushing states (`TRYUPGRADE` or `ACKUPGRADE`) passed in by caller, set the `FlushStatus` to `FLUSHING` and block sendpackets. During this time; `receivePacket`, `acknowledgePacket` and `timeoutPacket` will still be allowed and processed according to the original channel parameters. The new proposed upgrade will be stored in the public store for counterparty verification.
+`startFlushUpgradeHandshake` will set the counterparty last packet send and continue blocking the upgrade from continuing until all in-flight packets have been flushed. When the channel is in blocked mode, any packet receive above the counterparty last packet send will be rejected. It will verify the upgrade parameters and set the channel state to `FLUSHING` and block `sendPackets`. During this time; `receivePacket`, `acknowledgePacket` and `timeoutPacket` will still be allowed and processed according to the original channel parameters. The state machine will set a timer for how long the other side can take before it completes flushing and moves to `FLUSHCOMPLETE`. The new proposed upgrade will be stored in the public store for counterparty verification.
```

Comment:
I'd be in favour of this handler only aborting. That is, only cancel upgrades after you hit the flushing timer. If an upgrade is incompatible, abort and require either the authority of one chain to modify their proposed upgrade or require the counterparty chain to fast forward their sequence number